sycl/tools/sycl-xocc/bin/sycl-xocc

```diff
@@ -211,23 +211,23 @@ if [[ ! ${#KERNEL_NAME_ARRAY[@]} -eq 0 ]]; then
   # Compile all Kernels in the source file to seperate .xo ("object") files
   # and add them to the list of kernels to be linked
   LINKER_LIST=""
-  for index in "${KERNEL_NAME_ARRAY[@]}"; do
-  if [[ -n "$index" ]]; then
+  for ((i=0;i<${#KERNEL_NAME_ARRAY[@]};++i)); do
```

Comment:
What was the issue here?

Reply:
the is no issue in the previous implemenantation. but now we need to iterate on the KERNEL_NAME_ARRAY and EXTRA_KERNEL_PARAMS_ARRAY so we can't use the old system.